test_collections/sdk_tests/support/python_testing/models/test_case.py

```diff
@@ -116,14 +226,56 @@ def __class_name(identifier: str) -> str:
         """Replace all non-alphanumeric characters with _ to make valid class name."""
         return re.sub("[^0-9a-zA-Z]+", "_", identifier)
 
+    def run_command(self, cmd: str) -> None:
+        os.system(cmd)
+
+    async def execute(self) -> None:
+        try:
+            logger.info("Running Python Test: " + self.metadata["title"])
+            BaseManager.register("TestRunnerHooks", SDKPythonTestRunnerHooks)
+            manager = BaseManager(address=("0.0.0.0", 50000), authkey=b"abc")
+            manager.start()
+            test_runner_hooks = manager.TestRunnerHooks()  # type: ignore
+            runner_class = RUNNER_CLASS_PATH + RUNNER_CLASS
+            self.chip_tool.send_command(
+                f"{runner_class} {self.metadata['title']}", prefix=EXECUTABLE
+            )
+            while ((update := test_runner_hooks.updates_test()) is not None) or (
+                not test_runner_hooks.finished()
+            ):
+                if not update:
+                    continue
+
+                def handle_update(update: dict) -> None:
+                    def call_function(obj, func_name, kwargs) -> None:  # type: ignore
+                        func = getattr(obj, func_name, None)
+                        if not func:
+                            raise AttributeError(
+                                f"{func_name} is not a method of {obj}"
+                            )
+                        if not callable(func):
+                            raise TypeError(f"{func_name} is not callable")
+                        # Call the method with the unpacked keyword arguments.
+                        func(**kwargs)
+
+                    for func_name, kwargs in update.items():
+                        call_function(self, func_name, kwargs)
+
+                handle_update(update)
+        finally:
+            pass
+
+    async def cleanup(self) -> None:
+        logger.info("Test Cleanup")
+        self.chip_tool.destroy_device()
+
 
 class PythonChipToolTestCase(PythonTestCase, ChipToolTest):
-    """Automated Python test cases."""
+    """Automated Python test cases using chip-tool."""
 
     test_type = ChipToolTestType.PYTHON_TEST
 
     def create_test_steps(self) -> None:
-        self.test_steps = [TestStep("Start Python test")]
```

Comment:
Should this really be removed? I talked about this in another comment already and why I believe it should be kept